### PR TITLE
New Resource : azurerm_servicebus_namespace_disaster_recovery_config (Service Bus Geo DR)

### PR DESCRIPTION
**Pull request by 1-xtra**
As per issue #3136, implemented a disaster resource config resource, following the same pattern as the 'Event Hubs Geo DR config' (which is very similar)

* Added the DR client to the Service Bus clients
* Added parsers and validators for the resource id type
* Implemented resources + tests, following same patterns and naming conventions as per event hubs disaster config resource

Mainly putting this here for a bit of visibility and early feedback. Basic test is passing - still need to add the update tests

Any comments appreciated,

### CODE REVIEW

File: website/docs/r/servicebus_namespace_disaster_recovery_config.html.markdown [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  partner_namespace_id            = azurerm_resource_group.secondary.id
+}
+
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Disaster Recovery Config. This is the alias DNS name that will be created. Changing this forces a new resource to be created.
```
[the remark this answers is not in the document]
Comment (1-xtra) [reply website/docs/r/servicebus_namespace_disaster_recovery_config.html.markdown:52]: I initially went with **alias_name** as part of the schema as I thought it described the intention more accurately, but changed it to name to both align with the Event Hub schema for the DR config defined here (https://github.com/terraform-providers/terraform-provider-azurerm/blob/b6202bb72e3f108674b77f3d4f8c40c2d4ae1bbd/azurerm/internal/services/eventhub/eventhub_namespace_disaster_recovery_config_resource.go#L41) as well as following the consistent pattern that most resources seem to expose a 'name' field as part of the schema,

File: azurerm/internal/services/servicebus/servicebus_namespace_disaster_recovery_config_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	tags = {
+		Platform 	= "Platform Engineering"
+		Stage   	= "NonProd"
+	  }  	
+  }
+
+  resource "azurerm_servicebus_namespace_disaster_recovery_config" "pairing_test" {
+	name                    = "acctest-alias-%[1]d"
+	resource_group_name     = azurerm_resource_group.primary.name
+	namespace_name          = azurerm_servicebus_namespace.primary_namespace_test.name
```
[the remark this answers is not in the document]
Comment (1-xtra) [reply azurerm/internal/services/servicebus/servicebus_namespace_disaster_recovery_config_test.go]: done

File: website/docs/r/servicebus_namespace_disaster_recovery_config.html.markdown (hunk 1 above)
[the remark this answers is not in the document]
Comment (1-xtra) [reply website/docs/r/servicebus_namespace_disaster_recovery_config.html.markdown:51]: left as name - let me know thoughts

**Comment by 1-xtra** [comment]
Changes done : 

$ TF_ACC=1 go test -v ./azurerm/internal/services/servicebus -run=TestAccAzureRMServiceBusNamespacePairing_basic -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
2021/05/10 23:29:37 [DEBUG] not using binary driver name, it's no longer needed
2021/05/10 23:29:37 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccAzureRMServiceBusNamespacePairing_basic
=== PAUSE TestAccAzureRMServiceBusNamespacePairing_basic
=== CONT  TestAccAzureRMServiceBusNamespacePairing_basic
--- PASS: TestAccAzureRMServiceBusNamespacePairing_basic (914.15s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/servicebus  914.423s

File: website/docs/r/servicebus_namespace_disaster_recovery_config.html.markdown (hunk 1 above)
[the remark this answers is not in the document]
Comment (katbyte) [reply website/docs/r/servicebus_namespace_disaster_recovery_config.html.markdown:51]: I think both should be alias_name but given the other one is just name this'll do, maybe in 3.0 we can rename them to be a bit more clear

**Review by katbyte — APPROVED** [review]
Thanks @1-xtra - LGTM 👍

**Comment by 1-xtra** [comment]
Thanks @katbyte - my bad , didn't run-tflint - just done & shall be pushing a commit shortly that fixes the failed check. 

I have a vested interest in getting this over the line as a project I'm working on requires it. If anything needs more attention or polishing, just give me a mention & I'm happy to jump in.

**Pull request merged**